### ISSUE TITLE
also support javascript by default - Error: Qwik input "src/entry.ssr.tsx" not found.

### ISSUE THREAD

**Issue by milahu** [QwikDev/qwik#6100]
currently i have to add extra config to support javascript
but this should "just work" by default

vite.config.js

```js
export default defineConfig(({ command, mode }) => {
  return {
    plugins: [  
      qwikCity(),
      qwikVite({
        // fix: Error: Qwik input "src/entry.ssr.tsx" not found.
        //input: "src/entry.ssr.tsx", // default
        // https://github.com/BuilderIO/qwik/issues/6100
        // npm run build
        client: {
          input: "src/root.jsx",
        },
        // npm run dev
        ssr: {
          input: "src/entry.ssr.jsx",
        },
        dev: {
          input: "src/entry.dev.jsx",
        },
        // npm run build.preview
        preview: {
          input: "src/entry.preview.jsx",
        },
      }),
    ],
```

blame

https://github.com/BuilderIO/qwik/blob/7bc759c7ec00c197b9a546460eaad218ab6e43be/packages/qwik/src/optimizer/src/plugins/plugin.ts#L213

expected:
the plugin should check if `src/entry.ssr.tsx` exists
if yes, use it, if no, try `src/entry.ssr.jsx`

**Comment by wmertens** [comment]
PR welcome, and I suppose also for cjs, mjs, cts and mts. Perhaps with a readdir and using the first matching filename of a regex?

**Comment by ThatJSGuy** [comment]
I'd be happy to work on this fix if it's still needed. I'm looking to get more involved with the Qwik community, and this seems like a great place to start!

**Comment by PatrickJS** [comment]
@ThatJSGuy feel free to work on it and I can review it afterwards

**Comment by JerryWu1234** [comment]
https://github.com/JerryWu1234/qwik/blob/c99fec937a8f86ccfcbe3049dd7989d54f6c99b8/packages/qwik/src/optimizer/src/plugins/vite-server.ts#L234

```js
const entryPreviewPaths = ['mjs', 'cjs', 'js'].map((ext) =>
    path.join(ssrOutDir, `entry.preview.${ext}`)
  );

  const entryPreviewModulePath = entryPreviewPaths.find((p) => fs.existsSync(p));
  if (!entryPreviewModulePath) {
    return invalidPreviewMessage(
      middlewares,
      `Unable to find output "${ssrOutDir}/entry.preview" module.\n\nPlease ensure "src/entry.preview.tsx" has been built before the "preview" command.`
    );
  }
```

Perhas I could follow this step to develop this function

**Comment by JerryWu1234** [comment]
I'll pick this PR

**Comment by JerryWu1234** [comment]
@wmertens 
Do I only care about the **src/entry.ssr.jsx**?
how about `src/root.tsx`，`src/entry.dev.tsx` , `src/entry.preview.jsx`
Do I have to handle them?

**Comment by wmertens** [comment]
@JerryWu1234 for consistency, yes. However, I wonder what happens if you leave off the extension? Maybe vite resolves it by itself?

**Comment by JerryWu1234** [comment]
> @JerryWu1234 for consistency, yes. However, I wonder what happens if you leave off the extension? Maybe vite resolves it by itself?

I will prioritize doing this way `Maybe vite resolves it by itself?`
then other

**Comment by yousefmarey12** [comment]
I saw this as a good first issue and it's not closed. Is the issue not of concern anymore, or can I be an assignee?

**Comment by gioboa** [comment]
Hi @yousefmarey12  
it's closed by #6162
